lib/modules/datasource/endoflife-date/index.ts

```diff
@@ -0,0 +1,61 @@
+import { logger } from '../../../logger';
+import { cache } from '../../../util/cache/package/decorator';
+import { joinUrlParts } from '../../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { datasource, registryUrl } from './common';
+import type { EndoflifeDateVersion } from './types';
+
+export class EndoflifeDatePackagesource extends Datasource {
+  static readonly id = datasource;
+
+  override readonly defaultRegistryUrls = [registryUrl];
+  override readonly caching = true;
+  override readonly defaultVersioning = 'loose';
+
+  constructor() {
+    super(EndoflifeDatePackagesource.id);
+  }
+
+  @cache({
+    namespace: `datasource-${datasource}`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      // TODO: types (#7154)
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      `${registryUrl}:${packageName}`,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    if (!registryUrl) {
+      return null;
+    }
+
+    logger.trace(`${datasource}.getReleases(${registryUrl}, ${packageName})`);
+
+    const result: ReleaseResult = {
+      releases: [],
+    };
+
+    let response: { body: EndoflifeDateVersion[] };
+
+    const url = joinUrlParts(registryUrl, `${packageName}.json`);
+
+    try {
+      response = await this.http.getJson(url);
+
+      response.body.forEach((cycle) => {
```

Comment:
Use zod for ensuring the schema. https://github.com/renovatebot/renovate/blob/8b06cdc2cb38811e80280062552b85007ef65e95/lib/modules/manager/bazel/rules/docker.ts#L8

Reply:
@secustor I tried to update the code to use zod in e51c5efe3d5f461cd452f09ddc99f5745d5d13e6, but I seem to fail to understand something about how to use it. I took the existing code as a guideline, but since it's my first time using zod, I seem to miss something.
It's erroring with `Property 'version' is missing in type 'SafeParseSuccess<Release[]>' but required in type 'Release'.ts(2345)`.

From what I understood reading the documentation, the return value should be [these lines](https://github.com/morremeyer/renovate/blob/e51c5efe3d5f461cd452f09ddc99f5745d5d13e6/lib/modules/datasource/endoflife-date/types.ts#L28-L32), but apparently I'm wrong here?
It looks to me like it might be quite obvious to someone who knows zod better, so I pushed the broken state.

Could you take a look when you have time?

Reply:
You are mixing up Release objects and array of these. 

https://github.com/secustor/renovate/commit/02608b7f182cc040518b2edb94e4ec75fe4017e8